src/sentry/static/sentry/app/utils/discover/discoverQuery.tsx

```diff
@@ -87,18 +100,42 @@ class DiscoverQuery extends React.Component<Props, State> {
     );
   };
 
-  fetchData = () => {
-    const {eventView, orgSlug, location, limit, keyTransactions} = this.props;
+  getRoute(keyTransactions, trendsType) {
+    if (keyTransactions) {
+      return 'key-transactions';
+    }
+    if (trendsType) {
+      return 'events-trends';
+    }
+    return 'eventsv2';
+  }
 
-    if (!eventView.isValid()) {
+  fetchData = () => {
+    const {
+      eventView,
+      orgSlug,
+      location,
+      limit,
+      keyTransactions,
+      trendChangeType,
+    } = this.props;
+
+    if (!eventView.isValid() && !trendChangeType) {
```

Comment:
Why does using `trendChangeType` mean you can continue fetching data with an invalid `eventView`? Should this logic be lifted into `EventView` to make it valid?

Reply:
Because I didn't want to pollute the eventView if we planned to split off trends from discover query, which if trends gets significantly more complicated I plan to do. I suppose I could make a super set of the view and check that validity if it's a query from trends.

Reply:
To clarify a bit, a trends query will never pass this check because it never sends `fields` so an extended view would have an overridden function, which I think is more confusing as there then needs to be a trends view used everywhere in lieu of it. Would it helped if I added a comment to that effect here?

Reply:
Yeah a comment to call out the reason why this was done like this would be good here as its not immediately obvious.